command/debug.go

```diff
@@ -0,0 +1,709 @@
+	resp, err := client.Do(req)
+	c.writeBody(dir, "consul-agent-self.json", resp, err)
+
+	req, _ = http.NewRequest("GET", consul+"/v1/agent/members", nil)
+	req.Header.Add("X-Consul-Token", token)
+	req.Header.Add("User-Agent", userAgent)
+	resp, err = client.Do(req)
+	c.writeBody(dir, "consul-agent-members.json", resp, err)
+
+	return nil
+}
+
+// collectVault calls the Vault API directly to collect data
+func (c *DebugCommand) collectVault(dir, vault string) error {
+	if vault == "" {
+		return nil
+	}
+
+	token := c.vaultToken
+	if token == "" {
+		os.Getenv("VAULT_TOKEN")
+	}
+
+	client := http.Client{
+		Timeout: 2 * time.Second,
+	}
+
+	req, _ := http.NewRequest("GET", vault+"/sys/health", nil)
+	req.Header.Add("X-Vault-Token", token)
+	req.Header.Add("User-Agent", userAgent)
+	resp, err := client.Do(req)
+	c.writeBody(dir, "vault-sys-health.json", resp, err)
+
+	return nil
+}
+
+// writeBytes writes a file to the archive, recording it in the manifest
+func (c *DebugCommand) writeBytes(dir, file string, data []byte) error {
+	path := filepath.Join(dir, file)
+	c.manifest = append(c.manifest, path)
+	path = filepath.Join(c.collectDir, path)
+
+	fh, err := os.Create(path)
+	if err != nil {
+		return err
+	}
+	defer fh.Close()
+
+	_, err = fh.Write(data)
+	return err
+}
+
+// writeJSON writes JSON responses from the Nomad API calls to the archive
+func (c *DebugCommand) writeJSON(dir, file string, data interface{}) error {
+	bytes, err := json.Marshal(data)
+	if err != nil {
+		return err
+	}
+	return c.writeBytes(dir, file, bytes)
+}
+
+// writeBody is a helper that writes the body of an http.Response to the archive
+func (c *DebugCommand) writeBody(dir, file string, resp *http.Response, err error) {
+	if err != nil {
+		return
+	}
+	defer resp.Body.Close()
+
+	if resp.ContentLength == 0 {
+		return
+	}
+
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return
+	}
+
+	c.writeBytes(dir, file, body)
+}
+
+// writeManifest creates the index files
+func (c *DebugCommand) writeManifest() error {
+	// Write the JSON
+	path := filepath.Join(c.collectDir, "index.json")
+	json, err := os.Create(path)
+	if err != nil {
+		return err
+	}
+	defer json.Close()
+
+	json.WriteString("[\n")
+
+	first := true
+	for _, f := range c.manifest {
+		if first {
+			first = false
+			json.WriteString("  \"")
+		} else {
+			json.WriteString(",\n  \"")
+		}
+		json.WriteString(f)
+		json.WriteString("\"")
+
+	}
+
+	json.WriteString("\n]\n")
+
+	// Write the HTML
+	path = filepath.Join(c.collectDir, "index.html")
+	html, err := os.Create(path)
+	if err != nil {
+		return err
+	}
+	defer html.Close()
+
+	html.WriteString("<html><head><title>")
+	html.WriteString(c.timestamp)
+	html.WriteString("</title></head>\n<body>\n<ul>\n")
+
+	for _, f := range c.manifest {
+		html.WriteString("<li><a href=\"")
+		html.WriteString(f)
+		html.WriteString("\">")
+		html.WriteString(f)
+		html.WriteString("</a></li>\n")
+	}
+	html.WriteString("</ul>\n</body>\n</html>\n")
```

Comment:
Using `text/html` templates for this might be nicer, but not a blocker.

Reply:
👍